### PR TITLE
More Clean Up in test.sh

### PR DESCRIPTION
**Pull request by panpan0000**
(1) add more steps in clean_up when aborted (using signal EXIT).
(2) add cleanupVMs into end of test.sh.in (general_cleanup). it was not done when script normally exit.

@PengTian0  @changev  @iceiilin @anhou

### CODE REVIEW

**Comment by panpan0000** [comment]
test this please

File: test.sh.in [hunk 1]
```diff
@@ -346,6 +346,23 @@ fetchOVALog(){
...
+#
+#############################################
+general_cleanup(){
+
+    set +e
+    cleanupVMs
+    generateSysLog
+    generateMongoLog
+    vnc_record_stop
+
```
Comment (PengTian0) [inline test.sh.in]: The method ```cleanupVMs``` will destroy vagrant box.
The methods ```generateSysLog``` , ```generateMongoLog``` and ```vnc_record_stop``` all need to login the vagrant.
So, please run ```cleanupVMs``` at the last step.

Comment (panpan0000) [reply test.sh.in]: Good catch! fixed

File: test.sh.in [hunk 2]
```diff
@@ -346,6 +346,23 @@ fetchOVALog(){
...
+#  Ensure logs are dumpped even build timeout aborted
+#
+#############################################
+general_cleanup(){
+
+    set +e
+    generateSysLog
+    generateMongoLog
+    vnc_record_stop
+    cleanupVMs
```
Comment (PengTian0) [inline test.sh.in:361]: If we run ```cleanupVMs``` here, the feature "stop on failure" will be useless.

Comment (panpan0000) [reply test.sh.in:361]: Thank you , good catch, @PengTian0 

I've fixed it in https://github.com/RackHD/on-build-config/pull/195/commits/e80eb91f2dc2cfaeb2314e9fd868e03be1f20f7c

can you please review again ?

refer to article for trap usage : https://bash.cyberciti.biz/guide/How_to_clear_trap

**Comment by panpan0000** [comment]
test this please

**Comment by panpan0000** [comment]
@PengTian0 , can you please review again ?

**Comment by panpan0000** [comment]
this PR is no longer applicable, but the idea is still valid.
the cleanup should be taken care by signal handler.

**Comment by panpan0000** [comment]
@PengTian0 , I will close this PR.
But again, I still propose the cleanup being handle by signal-handler instead of in multiple places

**Pull request closed without merging**